sonar-plugin-api/src/main/java/org/sonar/api/server/rule/internal/DefaultRule.java

```diff
@@ -135,6 +137,11 @@ public List<RuleDescriptionSection> ruleDescriptionSections() {
     return Collections.unmodifiableList(ruleDescriptionSections);
   }
 
+  @Override
+  public Set<String> genericConceptKeys() {
+    return genericConceptKeys;
```

Comment:
return Collection.unmodifiableSet(genericConceptKeys)

Reply:
It's already done at constructor level